invenio_communities/members/services/components.py

```diff
@@ -14,32 +14,29 @@
 
 from invenio_communities.members.records.api import MemberMixin
 
-from ...utils import on_membership_change
+from ...proxies import current_identities_cache
+from ...utils import on_group_membership_change, on_user_membership_change
 
 
 class CommunityMemberCachingComponent(ServiceComponent):
     """Service component for community member caching."""
 
-    def _member_changed(self, member):
+    def _member_changed(self, member, community=None):
         """Call caching membership change function for user."""
 
-        def _get_user_ids_from_group_id(role_id):
-            r = Role.query.filter(Role.id == role_id).one()
-            return [u.id for u in r.users.all()]
-
         user_ids = []
         if isinstance(member, MemberMixin):
             if member.user_id:
                 user_ids = [member.user_id]
             elif member.group_id:
-                user_ids = _get_user_ids_from_group_id(member.group_id)
+                on_group_membership_change(str(community.id))
```

Comment:
major: `community` must be defined. Otherwise it can raise an exception. 


question: `member.group_id` wiped before calling this component?

Reply:
This function takes care of updating the members, for both `user` and `groups`. However, for users we do not need to pass the community but for groups we need. If we try to upgrade a group wihout passing the community it has to raise an exception. Alternatively we can split the functions to differentiate the change of membership between `users` and `groups`, but if we try to update a group without passing a community it will always raise an exception as it has to.

Reply:
> question: member.group_id wiped before calling this component?

As far as I have tested it's not wiped before calling the component. Why do you think it's being wiped before?

Reply:
I agree with Manuel, you have set the community to be None as default, maybe we need to raise an exception to check if it is passed? or in the conditional statement we need to check if community is not None?